lib/core/src/Cardano/Wallet.hs

```diff
@@ -2351,13 +2353,50 @@ derivePublicKey ctx wid role_ ix = db & \DBLayer{..} -> do
   where
     db = ctx ^. dbLayer @s @k
 
+-- | Retrieve public account key of a wallet.
+readPublicAccountKey
+    :: forall ctx s k n.
+        ( HasDBLayer s k ctx
+        , HardDerivation k
+        , WalletKey k
+        , s ~ SeqState n k
+        )
+    => ctx
+    -> WalletId
+    -> Passphrase "raw"
+    -> DerivationIndex
+    -> ExceptT ErrReadAccountPublicKey IO (k 'AccountK XPub)
+readPublicAccountKey ctx wid pwd ix = db & \DBLayer{..} -> do
+    acctIx <- withExceptT ErrReadAccountPublicKeyInvalidIndex $ guardHardIndex ix
```

Comment:
I wonder if from a user-experience perspective, it isn't better to only accept indexes in the soft range, and manually translate them to the hard range. So, when using the API, one would simply post `"index": 0` without having to worry about it being hardened or soft. 

Not sure if this is confusing or better :thinking:

Reply:
Ah, nevermind. Seeing the rest of the PR, I remembered that the JSON instance for `DerivationIndex` uses the `H` suffix. So account indexes are specified by users as `"index": "0H"` :+1: